src/frameworks/tide.rs

```diff
@@ -49,9 +49,10 @@ async fn create_restaurant(mut req: Request<State>) -> tide::Result {
     let repo = req.state().repo.clone();
     let runtime = req.state().runtime.clone();
     
-    let result = runtime.block_on(async move {
-        repo.create_restaurant(restaurant).await
-    });
+    let result = runtime
```

Comment:
Using `block_on` would cause the current thread to be, well, blocked until the handler finishes, which means depending on how it's implemented the async-std runtime (and by consequence Tide) might not be able to give time to other tasks.  `spawn` gives it its own parallel execution task so async-std/Tide can continue.  The `unwrap_or_else` is just merging in the potential "your task was cancelled" error from this into the normal error-handling path.